apps/hpc-ftsadmin/src/app/components/flow-form/flow-version.tsx

```diff
@@ -0,0 +1,180 @@
+import { t } from '../../../i18n';
+import { getContext } from '../../context';
+import paths from '../../paths';
+import { PENDING_REVIEW } from '../../utils/constants';
+import InfoAlert from '../info-alert';
+import FlowCompare, { type FlowVersion } from './flow-compare';
+
+type FlowVersionSelection = [FlowVersion | null, FlowVersion | null];
+
+const SPAN_STYLES = `
+  px-2
+  py-1
+  mx-2
+  border
+  border-solid
+  rounded-sm
+`;
+const LatestSpan = tw.span`
+  ${SPAN_STYLES}
+  bg-unocha-success-light
+  border-unocha-success
+`;
+const PendingReviewSpan = tw.span`
+  ${SPAN_STYLES}
+  bg-unocha-pallete-orange-light
+  border-unocha-pallete-orange
+`;
+const DeletedSpan = tw.span`
+  ${SPAN_STYLES}
+  bg-unocha-error-light
+  border-unocha-error
+`;
+const MarginEndLink = tw(Link)`
+  me-2
+`;
+const TableRowClick = tw(TableRow)`
+  transition-all
+  hover:shadow-md
+  bg-blend-hue
+  hover:bg-opacity-20
+  hover:cursor-pointer
+`;
+
+const FlowVersions = ({
+  flow,
+  inactiveReasons,
+}: {
+  flow: flows.GetFlowResult;
+  inactiveReasons: categories.Category[];
+}) => {
+  const { lang } = getContext();
+
+  const [selectedVersions, setSelectedVersions] =
+    useState<FlowVersionSelection>([null, null]);
+
+  const hasFlowsSelected = (
+    selection: FlowVersionSelection
+  ): selection is [FlowVersion, FlowVersion] =>
+    selection.every((flowVersion) => flowVersion !== null);
+
+  const pendingReviewCategory = inactiveReasons.find(
+    (category) => category.name === PENDING_REVIEW
+  );
+
+  const handleVersionSelection = (flowVersion: FlowVersion) => {
+    setSelectedVersions((prev) => {
+      const checkedRowIndex = prev.findIndex(
+        (a) => a?.id === flowVersion.id && a.versionID === flowVersion.versionID
+      );
+
+      if (checkedRowIndex !== -1) {
+        const prevSelectedRowsClone = structuredClone(prev);
+        prevSelectedRowsClone[checkedRowIndex] = null;
+        if (checkedRowIndex === 0) {
+          prevSelectedRowsClone[0] = prevSelectedRowsClone[1];
+          prevSelectedRowsClone[1] = null;
+        }
+        return prevSelectedRowsClone;
+      }
+
+      if (prev.every((selectedVersion) => selectedVersion !== null)) {
+        const prevSelectedRowsClone = structuredClone(prev);
+        prevSelectedRowsClone[1] = flowVersion;
+        return prevSelectedRowsClone;
+      }
+
+      if (!prev[0]) {
+        return [flowVersion, null];
+      }
+      const flowB = prev[1] ?? flowVersion;
+
+      return [prev[0], flowB];
+    });
+  };
+
+  return (
+    <Box sx={tw`flex flex-col px-4 gap-y-6`}>
+      <InfoAlert
+        text={t.t(lang, (s) => s.components.flowVersions.info.compareInfo)}
+        localStorageKey="compareFlowVersions"
+        sxProps={tw`mb-4`}
+      />
+      <Table>
+        <TableBody>
+          {flow.versions
+            ?.sort(
+              (flowVersion, previous) =>
+                previous.versionID - flowVersion.versionID
+            )
```

Comment:
`flow.versions` can only be an array, according to its type. Beware, sorting modifies the original array!

```suggestion
          {flow.versions
            .sort(
              (flowVersion, previous) =>
                previous.versionID - flowVersion.versionID
            )
```

Reply:
Would be nice to have ES2023 to do `toSorted()`. In this case I don't think it matters, but yeah I would prefer to not modify original array. Since we are not in ES2023, I dont think is worth doing a `structuredClone()` for a case that will not have impact, it would look uglier